File: predict.scala
import ml.dmlc.mxnet._
import ml.dmlc.mxnet.module.Module
import scala.collection.mutable.ArrayBuffer

val modelPrefix: String = "mx_mlp"
val modelEpoch: Int = 130

val mod = Module.loadCheckpoint(modelPrefix, modelEpoch, loadOptimizerStates = true)
//パラメータ確認
//val (argParams, auxParams) = mod.getParams

val dataiter = IO.ImageRecordIter(Map(
            "path_imgrec" -> "./test_v1.rec",
            "path_imglist" -> "./test_v1.lst",
            "data_shape" -> "(3,299,299)",
            "batch_size" -> "1"))

mod.bind(dataShapes = dataiter.provideData)
mod.initParams()

dataiter.reset()

var res = ArrayBuffer.empty[IndexedSeq[NDArray]]

while (dataiter.hasNext) {
    val dataBatch = dataiter.next()
    res += mod.predict(dataBatch)
}

val prob = res(0)
val arr = prob.toArray

File: predict.py
import mxnet as mx
import inception
import imageiter
import numpy as np
import os


model_prefix = 'mx_mlp'
num = 50
batch = 20

data_iter = imageiter.ImageIter(
    batch_size=batch,
    data_shape=(3, 299, 299),
    path_imgrec="./test.rec",
    path_imgidx="./test.idx",
    erasing=True 
)

sym, arg_params, aux_params = mx.model.load_checkpoint(model_prefix, num)

mod = mx.mod.Module(symbol=sym,
                    context=mx.cpu(),
                    data_names=['data'],
                    label_names=['softmax_label'])

mod.bind(for_training=False, data_shapes=[('data', (batch,3,299,299))])
mod.set_params(arg_params, aux_params)

y = mod.predict(data_iter)
print(y)

mx.nd.save('result.npz', y)
